OutSystems/outsystems-datagrid-reactive
# RGRIDT-697: ToOSFormat triggers an error with Date Columns

This PR is for RGRIDT-697: ToOSFormat triggers an error with Date Columns

### What was happening
* We were adding our columns to the grid.columns map in two different ways Map<binding,Column> and Map<uniqueID,Column>. This was indeed needed because we need to get the columns from the grid by both uniqueID and binding depending on the case. But it seems that it was generating two columns that were then added to the array of columns of the grid and it was causing some problems and one of them was the bug that was identified in this RGRIDT-697.
![70464928-a679-4073-b7c6-a0fddf20b4da](https://user-images.githubusercontent.com/6432232/111498455-e03e5880-8739-11eb-8a4a-0c1ec1e9ae40.gif)

The first time the method setDeepDate was invoked everything was fine and the date was being converted to the ISOstring format. So, because we had 2 columns (2 equal columns), the second time the method was being invoked, the format was breaking while trying to convert a ISOstring into an ISOstring.

### What was done
* Since we don't really want to have duplicated columns on the grid.columns map, the method that were adding both columns to the map was fixed.
* I created a new method to get the column by column binding, which first checks if the map has the unique ID (if provided) and then searches by the binding.

### Test Steps
1.On a grid with validations,
2.Edit the first cell from column Stock with the value 200.
Expected: The cell should have a red border and mark. On mouseover, a tooltip should show the message "The value should be smaller than 100."

1.On a grid with Date columns
2.Edit a cell from the Date column
3.Execute the code: GridAPI.GridManager.GetChangesInGrid(GridAPI.GridManager.GetActiveGrid()._uniqueId)
Expected: No errors should be triggered and the console should show the changes on the grid in a stringified format.


### Screenshots
**After the fix**
![formatDatesProblem](https://user-images.githubusercontent.com/6432232/111500094-876fbf80-873b-11eb-8e79-f7f35f259196.gif)


### Checklist
